db/migrate/20191101055132_backfill_granted_title_badge_id_for_user_profile.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+class BackfillGrantedTitleBadgeIdForUserProfile < ActiveRecord::Migration[6.0]
+  def up
+    # update all the regular badge derived titles based
+    # on the normal badge name
+    ActiveRecord::Base.connection.execute <<-SQL
+      UPDATE user_profiles
+      SET granted_title_badge_id = b.id
+      FROM users
+      INNER JOIN badges b ON users.title = b.name
+      WHERE users.title = b.name
+        AND user_profiles.granted_title_badge_id IS NULL
+        AND users.id = user_profiles.user_id
+        AND users.title IS NOT NULL AND users.title != ''
```

Comment:
I think we don't have to check the `users.title` here again, since we already checked it like `users.title = b.name`. If it's empty it won't even satisfy the existing inner join condition.